### ISSUE TITLE
[BUG] Documentation bug on state module virtualenv

### ISSUE THREAD

**Issue by sblaisot** [saltstack/salt#56729]
**Description**

Documentation of virtualenv state module https://docs.saltstack.com/en/master/ref/states/all/salt.states.virtualenv_mod.html refers to `virtualenv_mod` whereas it should be just `virtualenv` as correctly seen in the example state

**Comment by sagetherage** [comment]
@saltstack/docs-working-group

**Comment by sagetherage** [comment]
@barbaricyawps

**Comment by alexey-zhukovin** [comment]
https://github.com/saltstack/salt/blob/master/salt/modules/virtualenv_mod.py#L40-L44

Update the documentation:
replace `salt.states.virtualenv_mod` to `salt.states.virtualenv`

**Comment by sblaisot** [comment]
@alexey-zhukovin My bug was first targeted on state module instead of execution module, thus link to code is rather https://github.com/saltstack/salt/blob/master/salt/states/virtualenv_mod.py#L27-L34

But you're right, execution module is probably also concerned.

**Comment by sblaisot** [comment]
1+ year, still present.

Any news?

**Comment by waynew** [comment]
@sblaisot are you referring to, e.g. `salt.states.virtualenv_mod.manage(`
![screenshot of documentation showing salt.states.virtualenv_mod`](https://user-images.githubusercontent.com/189750/143083746-803fdf6a-747d-4c6f-9643-ae4cb3c70d78.png)

or something else?

That name is actually generated via sphinx and isn't really something we can control. Or are you referring to examples? Currently the only `virtualenv_mod` that I'm seeing in the state is the part generated via sphinx.

**Comment by sblaisot** [comment]
Yes, I'm talking about this sphinx part (in fact the difference between the filename and the state name)

It's pretty disturbing to see it referred as salt.states.virtualmodenv_mod. 

Most (if not all) other states are referred to as salt.states.<the state> and not something else.

You can also see that in the right pane it is referred as salt.states.virtualmod so there is something incoherent between the two.

**Comment by waynew** [comment]
It *might* be possible to use something like https://stackoverflow.com/questions/17078484/how-to-customize-module-name-for-sphinx 

I'm actually disinclined to say that we should rewrite the module though, since it wouldn't actually be available in Python via that path. I think what we *could* do is for modules where the `__virtualname__` is different than the actual module name, we could ensure that they have some kind of note in the heading.

The reason that I'm disinclined to say that this should be `.virtualenv.` instead of `.virtualenv_mod.` is that if we do *that* then we would either be inconsistent when it comes to other modules (e.g. `win_pkg`, `aptpkg` are available via `pkg`), or we would have to change the other modules as well, in a sort of ambiguous way.

---

I think a good idea would be:

```
.. note::

    This module have a virtual name, so even though it's defined as ``virtualenv_mod``, it will be called 
    via ``virtualenv`` -- see examples for correct usage.
```

Adding/requiring something like that at the top of the modules, etc. that have surprising names.
